### ISSUE TITLE
'await <T>(param) => body' visits body twice, causing duplicate errors

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#832]

**Comment by strager** [comment]
Hmm, I don't know what I was thinking when I filed this issue. `await <T>(param) => body` is not legal in TypeScript (inside or outside an async function). We seem to do an *okay* job at parsing it (certainly better than TS):
```
sandbox/hello.ts:1:1: error: 'await' cannot be followed by an arrow function; use 'async' instead [E0178]
sandbox/hello.ts:1:21: warning: use of undeclared variable: body [E0057]
sandbox/hello.ts:1:21: warning: use of undeclared variable: body [E0057]
```
Perhaps I meant the double E0057?

**Comment by strager** [comment]
Because the double visits is a symptom of possibly a deeper issue, I'm putting this in the 3.0 milestone.

**Comment by strager** [comment]
Fixed in commit 158256eb4b187e15e9d3797725e1619dc41111e4. The deeper issue was #831.
